src/Chainweb/WebPactExecutionService.hs

```diff
@@ -41,27 +41,54 @@ import Pact.Types.Hash
 -- -------------------------------------------------------------------------- --
 -- PactExecutionService
 
+-- | Service API for interacting with a single or multi-chain ("Web") pact service.
+-- Thread-safe to be called from multiple threads. Backend is queue-backed on a per-chain
+-- basis.
 data PactExecutionService = PactExecutionService
-    { _pactValidateBlock :: BlockHeader -> PayloadData -> IO PayloadWithOutputs
-    , _pactNewBlock :: Miner -> ParentHeader -> IO PayloadWithOutputs
-    , _pactLocal :: ChainwebTransaction -> IO (Either PactException (CommandResult Hash))
-    , _pactLookup
-        :: Rewind
-            -- restore point. 'NoRewind' means we
-            -- don't care about the restore point.
+    { _pactValidateBlock :: !(
+        BlockHeader ->
+        PayloadData ->
+        IO PayloadWithOutputs
+        )
+      -- ^ Validate block payload data by running through pact service.
+    , _pactNewBlock :: !(
+        Miner ->
+        ParentHeader ->
+        IO PayloadWithOutputs
+        )
+      -- ^ Request a new block to be formed using mempool
+    , _pactLocal :: !(
+        ChainwebTransaction ->
+        IO (Either PactException (CommandResult Hash)))
+      -- ^ Directly execute a single transaction in "local" mode (all DB interactions rolled back).
+      -- Corresponds to `local` HTTP endpoint.
+    , _pactLookup :: !(
+        Rewind
+        -- ^ restore point, either a block header or the current "head" of the pact service.
         -> Vector PactHash
-            -- txs to lookup
+        -- ^ txs to lookup
         -> IO (Either PactException (Vector (Maybe (T2 BlockHeight BlockHash))))
-    , _pactPreInsertCheck
-        :: ChainId
+        )
+      -- ^ Lookup pact hashes as of a block header to detect duplicates
+    , _pactPreInsertCheck :: !(
+        ChainId
         -> Vector ChainwebTransaction
-        -> IO (Either PactException (Vector (Either InsertError ())))
+        -> IO (Either PactException (Vector (Either InsertError ()))))
+      -- ^ Run speculative checks to find bad transactions (ie gas buy failures, etc)
+    , _pactBlockTxHistory :: !(
```

Comment:
A quick comment on the purpose of `_pactBlockTxHistory` would be helpful, like the comment in the other functions of `PactExecutionService`